YSI_Game/y_weapondata/y_weapondata_entry.inc

```diff
@@ -502,67 +504,294 @@ stock bool:Weapon_GetProAnimGroup(weaponid, dest[], size = sizeof (dest))
 
 stock bool:Weapon_IsAssaultRifle(weaponid)
 {
-	switch(weaponid)
-	{
-		case 30, 31: return true;
-	}
-	return false;
+    switch(weaponid)
+    {
+        case 30, 31: return true;
+    }
+    return false;
 }
 
 stock bool:Weapon_IsPistol(weaponid)
 {
-	switch(weaponid)
-	{
-		case 22, 23, 24: return true;
-	}
-	return false;
+    switch(weaponid)
+    {
+        case 22, 23, 24: return true;
+    }
+    return false;
 }
 
 stock bool:Weapon_IsShotgun(weaponid)
 {
-	switch(weaponid)
-	{
-		case 25, 26, 27: return true;
-	}
-	return false;
+    switch(weaponid)
+    {
+        case 25, 26, 27: return true;
+    }
+    return false;
 }
 
 stock bool:Weapon_IsSubmachineGun(weaponid)
 {
-	switch(weaponid)
-	{
-		case 28, 29, 32: return true;
-	}
-	return false;
+    switch(weaponid)
+    {
+        case 28, 29, 32: return true;
+    }
+    return false;
 }
 
 stock bool:Weapon_IsRifle(weaponid)
 {
-	switch(weaponid)
-	{
-		case 33, 34: return true;
-	}
-	return false;
+    switch(weaponid)
+    {
+        case 33, 34: return true;
+    }
+    return false;
 }
 
 stock Weapon_GetSlot(weaponid)
 {
-	return Weapon_IsValid(weaponid) ? YSI_g_WeaponSlots[weaponid] : (WEAPONDATA_INVALID_RETURN);
+    return (weaponid >= 1 && weaponid <= 18 || weaponid >= 21 && weaponid <= 46) ? YSI_g_WeaponSlots[weaponid] : (WEAPONDATA_INVALID_RETURN);
```

Comment:
It should return UNKNOWN_WEAPON_SLOT if the user is using `omp-stdlib`. It happens to be the same value currently but it would be safer to use the same definition.
Also the function (this and all others actually) should be tagged

Reply:
There actually is a native `GetWeaponSlot` that's built-in